lua/luasnip/session/snippet_collection.lua

```diff
@@ -202,43 +202,72 @@ local current_id = 0
 -- snippets like {ft1={<snippets>}, ft2={<snippets>}}, opts should be properly
 -- initialized with default values.
 function M.add_snippets(snippets, opts)
-	local prio_snip_table = by_prio[opts.type]
-
 	for ft, ft_snippets in pairs(snippets) do
-		local ft_table = by_ft[opts.type][ft]
-
-		if not ft_table then
-			ft_table = {}
-			by_ft[opts.type][ft] = ft_table
-		end
-
-		-- TODO: not the nicest loop, can it be improved? Do table-checks outside
-		-- it, preferably.
+		local prios = {
+			autosnippets = {},
+			snippets = {},
+		}
+		local types = {
+			autosnippets = false,
+			snippets = false,
+		}
+
+		-- collect which tables should be added
+		-- and do some initialization
 		for _, snip in ipairs(ft_snippets) do
 			snip.priority = opts.override_priority
 				or (snip.priority ~= -1 and snip.priority)
 				or opts.default_priority
 				or 1000
 
-			if not prio_snip_table[snip.priority] then
-				prio_snip_table[snip.priority] = {}
-			end
+			-- TODO
+			-- prefer more specific option I guess.
```

Comment:
Not quite sure what you mean by this comment. Does this refer to the name of the option?

Reply:
Oh, I meant that, since autotriggered can be set (or unset!) on the snippet and `add_snippets` (`type = "autosnippets"/type="snippets"`), to potentially conflicting values, we should prefer using the snippet-value since it's more specific.

Reply:
Ah I understand, that was previously resolved by your (if then else) expression in the next line, and still should be solved by the similar expression (just ported it to `-1`/`0`/`1` values)

Reply:
IMO it would be a tiny bit nicer if the snippet, at the end, had true/false-values again.
But that's really elaborate, I think the proper solution would be making `snippetProxy` not `initialize` on some values, and just return nil if they are nil.
That way we can use more sensible values in prio and autotriggered.

Reply:
I implemented that suggestion in `sProxy_allow_nil_select`, I'll merge that into master later I think.

Reply:
> at the end, had true/false-values again.

Well, yes this would be nicer, but only if we can use booleans in the first place as well (i.e. never use -1/0/1), as having the same key populated with different data types (int/bool) over time is kinda confusing, I think.

Is this (always using bools like in the beginning of this PR) possible with the changes you've made? (I'm not into `snippetProxy`)

Reply:
Bools + nil is possible, yes👍👍
I'll merge it into master in a bit.

Reply:
(you could also cherry-pick it into this branch, but I think putting it into master + rebase this PR will end up cleaner)

Reply:
Even tough having the chance of playing with cherry-pick (never used it (never had to) in reality) is tempting, merging the master into this branch is definitely the cleaner way (rebasing/rewriting history isn't that nice since I already "published" the branch here).
So this is done. Thanks for implementing in `snippetProxy` :+1:

Reply:
Feel free to rebase (in other instances, merge is fine), I can see your point from a technical POV, but imo it's not necessary to limit oneself in this way :sweat_smile:

Reply:
(missclick :sweat_smile:)